lib/kernel-manager.coffee

```diff
@@ -98,7 +74,13 @@ class KernelManager
 
 
     getAllKernelSpecs: ->
-        return _.map @_kernelSpecs, 'spec'
+        # TODO(nikita): switch to async and jupyter-js-services API
+        # Note that using sync here is a temporary regression, because it forces
+        # a manual call to update-kernels by the user.
+        # This will be fixed in a later patch, that adds proper async back
+        kernelSpecs = zmq_services.getZMQGateway().getKernelSpecsSync()
+        _.forEach kernelSpecs, (obj, name) -> obj.spec.name = name
```

Comment:
Can we call it something less ambiguous?
I don't want people to start using `spec.name` for anything else than spawning a gateway kernel.
Perhaps `spec._name`?